### DISCUSSION TITLE
Full Text Search

### DISCUSSION THREAD

**Discussion by kairan77** [singerdmx/flutter-quill#702] [General]
Is it possible for quill to support full text search within the editor? 

Use case:
- User enter keywords in a search box, matching occurrences of the search keywords are highlighted in the editor

@singerdmx, any idea on 

1. whether this is possible theoretically 
2. best place/api/class to hook up and implement such a function
3. pitfalls/alternatives

Thanks

**Comment by singerdmx** [comment]
you are talking about fuzzy search ?

**Comment by singerdmx** [comment]
Dumb idea is to go through all underlying delta data and do a string match

**Comment by singerdmx** [comment]
Planning to add this. I am not a good PM so any suggestion regarding UI and the workflow is welcome

**Comment by singerdmx** [comment]
I think we should expose a public function that users can call, which takes in the string and return a list of offset that matches the string. Also I think we already have function to move cursor to target offset. So it is up to users to do whatever with the UI.
So UI will not be part of the implementation.

**Comment by singerdmx** [comment]
I have submitted https://github.com/singerdmx/flutter-quill/commit/d2b308dc4d4c70620991185bbd1d0dd0911b69a5

**Comment by singerdmx** [comment]
Search feature released in 5.3.0
